### PR TITLE
Improve no resource found error messages for `get` command

### PR DESCRIPTION
**Pull request by kandros**
/kind cleanup

**What this PR does / why we need it**:
explicit error message to make it easier to notice when the user is searching resources in the wrong place

**Does this PR introduce a user-facing change?**:

```release-note
"no resource found" error messages for `get` command now mention the kind of resource and the namespace in which the command looked for
```

### CODE REVIEW

File: pkg/kubectl/cmd/get/get.go [hunk 1]
```diff
@@ -570,7 +570,10 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
 	w.Flush()
 	if trackingWriter.Written == 0 && !o.IgnoreNotFound && len(allErrs) == 0 {
 		// if we wrote no output, and had no errors, and are not ignoring NotFound, be sure we output something
-		fmt.Fprintln(o.ErrOut, "No resources found.")
+
```
Comment (draveness) [inline pkg/kubectl/cmd/get/get.go]: small nit: extra line

**Comment by kandros** [comment]
I signed it

File: pkg/kubectl/cmd/get/get.go [hunk 2]
```diff
@@ -570,7 +570,10 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
...
+		resourceType := args[0]
```
Comment (draveness) [inline pkg/kubectl/cmd/get/get.go]: Does this argument always exist when calling this cmd?

File: pkg/kubectl/cmd/get/get.go [hunk 3]
```diff
@@ -570,7 +570,10 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
...
+		fmt.Fprintf(o.ErrOut, "No %s found in namespace %s.\n", resourceType, o.Namespace)
+
```
Comment (draveness) [inline pkg/kubectl/cmd/get/get.go]: nit: extra line

File: pkg/kubectl/cmd/get/get_test.go [hunk 4]
```diff
@@ -352,10 +352,25 @@ foo    0/0              0          <unknown>   <none>
...
-	tf := cmdtesting.NewTestFactory().WithNamespace("test")
-	defer tf.Cleanup()
-
-	emptyTable := ioutil.NopCloser(bytes.NewBufferString(`{
+	testData := []struct {
+		cmdArg           string
+		defaultNamespace string
+		expectedErr      string
+	}{
+		{cmdArg: "pods", defaultNamespace: "test", expectedErr: `No pods found in namespace test.
```
Comment (draveness) [inline pkg/kubectl/cmd/get/get_test.go]: IMO the following approach is preferred:

```go
{
   cmdArgs: "pods"
   ...
}
```

**Review by draveness — CHANGES_REQUESTED** [review]
/ok-to-test
/priority backlog

**Comment by draveness** [comment]
Please sign the CLA following the instructions which the robot posted above.

File: pkg/kubectl/cmd/get/get.go (hunk 2 above)
Comment (kandros) [reply pkg/kubectl/cmd/get/get.go]: yes, it's always the resource type, when not provided it gets handled before this code at https://github.com/kubernetes/kubernetes/blob/master/pkg/kubectl/cmd/get/get.go#L272

if the resource-type is combined with a resource-name (eg, "pod/mypod") it also get handled already

File: pkg/kubectl/cmd/get/get.go [hunk 5]
```diff
@@ -570,7 +570,8 @@ func (o *GetOptions) Run(f cmdutil.Factory, cmd *cobra.Command, args []string) e
 	w.Flush()
 	if trackingWriter.Written == 0 && !o.IgnoreNotFound && len(allErrs) == 0 {
 		// if we wrote no output, and had no errors, and are not ignoring NotFound, be sure we output something
-		fmt.Fprintln(o.ErrOut, "No resources found.")
+		resourceType := args[0]
+		fmt.Fprintf(o.ErrOut, "No %s found in namespace %s.\n", resourceType, o.Namespace)
```
Comment (oomichi) [inline pkg/kubectl/cmd/get/get.go]: After merging https://github.com/kubernetes/kubernetes/commit/b7d2478ac9875f26250ea0803d92b58fe543e4a1 the output of kubectl contains the namespace like `No resources found in %s namespace.`
So now this PR adds a resource name into the error message.
From user viewpoint, they know what they specify as a resource.
Do we need to add it?

I guessed `$ kubectl get all` could show this error and this change could be useful to know what actual resources don't exist. However `$ kubectl get all` doesn't show such resources like the following:
```
$ kubectl get all
NAME                 TYPE        CLUSTER-IP   EXTERNAL-IP   PORT(S)   AGE
service/kubernetes   ClusterIP   10.96.0.1    <none>        443/TCP   5d6h
```
So I don't see actual merit at this time.

**Pull request closed without merging**